src/component/parser/json/converter/ProcessConverter.ts

```diff
@@ -88,21 +95,54 @@ export default class ProcessConverter extends AbstractConverter<Process> {
   // eslint-disable-next-line @typescript-eslint/no-explicit-any
   private buildFlowNodeBpmnElements(processId: string, bpmnElements: Array<any> | any, kind: ShapeBpmnElementKind): void {
     ensureIsArray(bpmnElements).forEach(bpmnElement => {
-      if (kind as BpmnEventKind) {
-        // get the list of eventDefinition hold by the Event bpmElement
-        const eventDefinitions = Object.values(EventDefinition).filter(eventDefinition => {
-          return bpmnElement.hasOwnProperty(eventDefinition);
-        });
-        // do we have a None Event?
-        if (eventDefinitions.length == 0) {
-          convertedFlowNodeBpmnElements.push(new ShapeBpmnElement(bpmnElement.id, bpmnElement.name, kind, processId));
+      if (this.isEvent(kind)) {
+        const shapeBpmnEvent = this.buildShapeBpmnEvent(bpmnElement, kind, processId);
+        if (shapeBpmnEvent) {
+          convertedFlowNodeBpmnElements.push(shapeBpmnEvent);
         }
       } else {
         convertedFlowNodeBpmnElements.push(new ShapeBpmnElement(bpmnElement.id, bpmnElement.name, kind, processId));
       }
     });
   }
 
+  private isEvent(kind: ShapeBpmnElementKind): boolean {
+    return [ShapeBpmnElementKind.EVENT_START, ShapeBpmnElementKind.EVENT_END].includes(kind);
+  }
+
+  // eslint-disable-next-line @typescript-eslint/no-explicit-any
+  private buildShapeBpmnEvent(bpmnElement: any, elementKind: ShapeBpmnElementKind, processId: string): ShapeBpmnEvent | void {
+    const eventDefinitions = this.getEventDefinitions(bpmnElement);
+    const numberOfEventDefinitions = eventDefinitions.map(eventDefinition => eventDefinition.counter).reduce((counter, it) => counter + it, 0);
+
+    // do we have a None Event?
+    if (numberOfEventDefinitions == 0) {
+      return new ShapeBpmnEvent(bpmnElement.id, bpmnElement.name, elementKind, ShapeBpmnEventKind.NONE, processId);
+    }
+
+    if (numberOfEventDefinitions == 1) {
+      const eventDefinition = eventDefinitions.filter(eventDefinition => eventDefinition.counter == 1)[0];
+      if (supportedBpmnEventKinds.includes(eventDefinition.kind)) {
+        return new ShapeBpmnEvent(bpmnElement.id, bpmnElement.name, elementKind, eventDefinition.kind, processId);
+      }
+    }
+  }
+
+  /**
+   * Get the list of eventDefinition hold by the Event bpmElement
+   *
+   * @param bpmnElement The BPMN element from the XML data which represents a BPMN Event
+   */
+  // eslint-disable-next-line @typescript-eslint/no-explicit-any
+  private getEventDefinitions(bpmnElement: any): EventDefinition[] {
+    return bpmnEventKinds.map(eventKind => {
+      // sometimes eventDefinition is simple and therefore it is parsed as empty string "", in that case eventDefinition will be converted to an empty object
+      const eventDefinition =
+        bpmnElement[eventKind + 'EventDefinition'] !== undefined ? (bpmnElement[eventKind + 'EventDefinition'] === '' ? {} : bpmnElement[eventKind + 'EventDefinition']) : null;
```

Comment:
To have a code more readable, I would extract **bpmnElement[eventKind + 'EventDefinition']** in a variable ;)